File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/java/src/empresa/Estagiario.java
package empresa;

import java.time.LocalDate;

public class Estagiario extends Funcionario {
    private int cargaHoraria;
    private String supervisor;

    public Estagiario(String nome, String cpf, double salarioBase, Departamento departamento,
                        LocalDate dataAdmissao, int cargaHoraria, String supervisor) {
        super(nome, cpf, salarioBase, departamento, dataAdmissao);
        this.cargaHoraria = cargaHoraria;
        this.supervisor = supervisor;
    }

    public int getCargaHoraria() {
        return cargaHoraria;
    }

    public String getSupervisor() {
        return supervisor;
    }

    @Override
    public double calcularBonus() {
        return salarioBase * 0.02;
    }

    @Override
    public double calcularSalarioTotal() {
        return salarioBase + calcularBonus();
    }

    @Override
    public String toString() {
        return "[EST] " + super.toString() + " | Carga: " + cargaHoraria + "h | Supervisor: " + supervisor;
    }
}

File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/dart/empresa/estagiario.dart
class Estagiario extends Funcionario {
  int cargaHoraria;
  String supervisor;

  Estagiario({
    required String nome,
    required String cpf,
    required double salarioBase,
    required Departamento departamento,
    required DateTime dataAdmissao,
    required this.cargaHoraria,
    required this.supervisor,
  }) : super(
          nome: nome,
          cpf: cpf,
          salarioBase: salarioBase,
          departamento: departamento,
          dataAdmissao: dataAdmissao,
        );

  @override
  double calcularBonus() => salarioBase * 0.02;

  @override
  double calcularSalarioTotal() => salarioBase + calcularBonus();

  @override
  String toString() {
    return "[EST] ${super.toString()} | Carga: ${cargaHoraria}h | Supervisor: $supervisor";
  }
}